designs/plugin-resolution.md

```diff
@@ -0,0 +1,87 @@
+# Plugin and project version resolution
+
+## Old behavior
+
+The same process is followed for all commands, without distinction between `init` and the rest. The only difference is
+that, for `init`, the configuration file will not be present.
+
+1. Obtain the project version from project configuration file's `version` field, `--project-verison` flag, or default
+   project version configured during CLI creation (`cli.New(cli.WithDefaultProjectVersion(...))`).
+    * In case both a project configuration file and the flag are found, fail if they are different.
+2. Obtain the plugin keys from the project configuration file's `layout` field, `--plugins` flag, or default plugins
+   configured during CLI creation for the above project version (`cli.New(cli.WithDefaultPlugins(...))`).
+    * In case both a project configuration file and the flag are found, fail if they are different.
+3. In case any of the plugin keys is not fully qualified (full name and version), check if there is only one plugin that
+   fits that unqualified plugin key.
+4. Resolve the plugins with the list of qualified keys above.
+5. Verify the plugins support the project version.
+   
+
+### Flaws
+
+* Default plugins need to be provided per project version
+* Project version can't be resolved (e.g., when a single project version is supported by the provided plugins).
+* Command calls are not allowed to override plugin keys.
+
+## Proposal
+
+The following plugin resolution algorithm solves the flaws described above:
+
+1. [All commands but Init] Obtain the project version and plugin keys from the project configuration if available.
+     * If they are available, plugin keys will be fully qualified.
+2. [Init command only] Obtain the project version from `--project-version` flag.
```

Comment:
One of the advantages implementation-wise of this new approach is that `--project-version` stops being a global variable for all commands that need to be special cased. It will just be a local variable of the `Init` command.